subsys/dsp/Kconfig

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2022 Google LLC
+# SPDX-License-Identifier: Apache-2.0
+
+config DSP_BACKEND_HAS_STATIC
+	bool
+
+choice DSP_BACKEND
+	prompt "DSP library backend selection"
+	default DSP_BACKEND_CMSIS if CMSIS_DSP
+	default DSP_BACKEND_THIRD_PARTY
+
+config DSP_BACKEND_CMSIS
+	bool "Use the CMSIS-DSP library as the math backend"
+	depends on CMSIS_DSP
+	select DSP_BACKEND_HAS_STATIC
+	help
+	  Implement the various zephyr DSP functions using the CMSIS-DSP library. This feature
+	  requires the CMSIS module to be selected.
+
+config DSP_BACKEND_THIRD_PARTY
+	bool "Do not use any Zephyr backends for DSP"
+	help
+	  Rely on the application to provide a DSP backend. The implementation should be added to
+	  the 'zdsp' build target by the application or one of its modules.
```

Comment:
> The implementation should be added to the 'zdsp' build target by the application

This sounds more like a "first party" than a "third party."

I wonder if it would be better to name this `DSP_BACKEND_NONE`?

Reply:
I see what you mean, I'm OK with either name. I would think that `NONE` would mean it's also not needed, but in this case you'd get an error. So `THIRD_PARTY` was meant to say I'm choosing a non-Zephyr backend (I'm bringing in my own).

Reply:
@stephanosio I left this for now to give you a change to read my reasoning for the term `THIRD_PARTY`, let me know which way you want to go with it. I feel like `NONE` is misleading because the application would still need to provide one so maybe we can come up with a better term?

Reply:
On second thought, I agree `NONE` can be also quite misleading.

What do you think about `CUSTOM` then? It is a "custom backend" in the sense that it is not one of the backends provided by the DSP subsystem and user has to provide their own **custom** backend.

Reply:
Done